File: backend/java/spring/rf-s-core-jdbc/src/main/java/com/rfindustries/corejdbc/dto/BaseJDBCDTO.java
package com.rfindustries.corejdbc.dto;

import com.rfindustries.core.dto.BaseDTO;
import lombok.Data;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@Data
@SuperBuilder
public class BaseJDBCDTO implements BaseDTO {
    private Long id;
    private Integer userCreatedAtId;
    private Integer userUpdatedAtId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private BaseJDBCUserDTO userCreatedAt;
    private BaseJDBCUserDTO userUpdatedAt;

    @SuppressWarnings("unchecked")
    @Override
    public <T> T resolverUserCreatedAtId() {
        return (T) (userCreatedAt != null && userCreatedAt.getId() != null ? userCreatedAt.getId() : this.userCreatedAtId);
    }

    @SuppressWarnings("unchecked")
    @Override
    public <T> T resolverUserUpdatedAtId() {
        return (T) (userUpdatedAt != null && userUpdatedAt.getId() != null ? userUpdatedAt.getId() : this.userUpdatedAtId);
    }
}

File: backend/go/rf-core/pkg/dto/basejdbc_dto.go
package dto

import "time"

type BaseJDBCDTO struct {
	BaseDTO
	Id                 int64     `json:"id"`
	UserCreatedAtId    int64     `json:"userCreatedAtId"`
	UserUpdatedAtId    int64     `json:"userUpdatedAtId"`
	CreatedAt          time.Time `json:"createdAt"`
	UpdatedAt          time.Time `json:"updatedAt"`
	BusinessCustomerId int64     `json:"businessCustomerId"`
	EnterpriseId       int64     `json:"enterpriseId"`
}
